service/history/visibilityQueueTaskExecutor.go

```diff
@@ -378,6 +378,10 @@ func (t *visibilityQueueTaskExecutor) processCloseExecution(
 		searchAttr,
 		historySizeBytes,
 	)
+	if err != nil {
+		return err
+	}
+	return t.cleanupExecutionInfo(ctx, task)
```

Comment:
`cleanupExecutionInfo` takes the lock and cleans up the mutable state. I wonder what if the clean up returns an err...